Core/src/IfcGeometryConverter/GeomUtils.h

```diff
@@ -62,93 +62,286 @@ namespace OpenInfraPlatform
+				/*! \brief Calculates a closest point on the line to the point. 
+				*
+				* \param[in] point					Point outside the line. 
+				* \param[in] line_origin			Line origin point. 
+				* \param[in] line_direction			Direction of the line. 	
+				*
+				* \param[out] closest				Closest point on the line to the point.
+				*/
 				static void closestPointOnLine(const carve::geom::vector<3>& point,
 					const carve::geom::vector<3>& line_origin,
 					const carve::geom::vector<3>& line_direction,
 					carve::geom::vector<3>& closest);
 
-
+				/**********************************************************************************************/
+				/*! \brief Calculates a closest point on the line to the point.
+				*
+				* \param[in] point					Point outside the line.
+				* \param[in] line_origin			Line origin point.
+				* \param[in] line_direction			Direction of the line.
+				*
+				* \param[out] closest				Closest point on the line to the point.
+				*/
 				static void closestPointOnLine(const buw::Vector3f& point,
 					const buw::Vector3f& line_origin,
 					const buw::Vector3f& line_direction,
 					buw::Vector3f& closest);
 
+				/**********************************************************************************************/
+				/*! \brief Verify if point is on the segment of the line. 
+				*
+				* \param[in] lambda					Lambda of the line. 
+				* \param[in] point					Point which should be verified. 
+				* \param[in] line_origin			Line origin point.
+				* \param[in] line_direction			Direction of the line.
+				*
+				* \return 							Returns true, if point is on the line. False otherwise.
+				*/
 				static bool isPointOnLineSegment(double& lambda,
 					const buw::Vector3f& point,
 					const buw::Vector3f& line_origin,
 					const buw::Vector3f& line_direction);
-
+				
+				/**********************************************************************************************/
+				/*! \brief Extrude geometry. 
+				*
+				* \param[in] paths					List of several sets of points of the geometry.
+				* \param[in] dir					The direction, where geometry should be extruded.
+				* \param[in] closed					Boolean operator, which shows, if geometry is closed.
+				*
+				* \param[out] poly_data				List of faces of the geometry.
+				*/
 				static void extrude(const std::vector<std::vector<carve::geom::vector<2>>>& paths,
 					const carve::geom::vector<3> dir,
 					const bool closed,
 					std::shared_ptr<carve::input::PolyhedronData>& poly_data);
 
+				/**********************************************************************************************/
+				/*! \brief Incorporates voids to the geometry. 
+				*
+				* \param[in] paths					List of several sets of points. 
+				*
+				* \param[out] merged_path			Arranged list of points.
+				* \param[out] triangulated			List of triangles to visualize geometry.
+				* \param[out] path_all_loops		List of several sets of indices. 
+				*/

```

Comment:
I am not sure, if this description is correct. Can you please give me a feedback regarding this doxy-comment

Reply:
We need to talk about this one.